apps/src/code-studio/assets/SpriteUpload.js

```diff
@@ -1,18 +1,52 @@
 import React from 'react';
 
 export default class SpriteUpload extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      file: null
-    };
-  }
+  state = {
+    fileData: null,
+    filePreviewURL: '',
+    filename: '',
+    category: '',
+    uploadState: ''
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
+
+    return fetch(
+      `/api/v1/animation-library/level_animations/${this.state.filename}`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'image/png'
+        },
+        body: this.state.fileData
+      }
+    )
+      .then(response => {
+        this.setState({
+          uploadState: response.ok
```

Comment:
nit: "uploadState" feels redundant here -- maybe "uploadStatus"?

i mentioned this below, but i think we should distinguish between success and error in this state for styling purposes. so this state could be something like:

```javascript
uploadStatus: {
  success: false, // property could be named "ok" or similar
  message: ''
}
```